python/tskit/metadata.py

```diff
@@ -544,8 +544,14 @@ def __eq__(self, other) -> bool:
 
     @property
     def schema(self) -> Optional[Mapping[str, Any]]:
-        # Make schema read-only
-        return self._schema
+        # Return a copy to avoid unintentional mutation
+        return copy.deepcopy(self._schema)
+
+    def asdict(self) -> Optional[Mapping[str, Any]]:
+        """
+        Return a deep copy of the schema.
```

Comment:
This isn't a super descriptive comment - we should be distinguishing the schema as an object that we're modelling (which is ``self``) and the dictionary representation of the schema. This was one of the main things that confused me when getting my head around the API. The fact we're storing the dictionary representation of the schema in a public attribute called ``schema`` is a detail that people shouldn't need to know about.

Reply:
Yes! I've hopefully improved it.